### PR TITLE
Don't log BuildEnvironmentWarning as error

### LINKED ISSUE #4590: Cleanup exception that are not logged as error
After #4495 got merged @agjohnson suggested to have an attribute in the Exception class and check for that attribute before log the exception, instead of defining a list for the warning exceptions as I did at:

https://github.com/rtfd/readthedocs.org/pull/4495/files#diff-ca52b098301dd315a834b3556ab9a7d5R424

Also, there are more exceptions that have to treat in the same way: `ProjectConfigurationError` for example.

https://sentry.io/read-the-docs/readthedocs-org/issues/668248681/

### PR DESCRIPTION
**Pull request by stsewd**
Turns out we are logging this manually
https://github.com/readthedocs/readthedocs.org/blob/00ab116ec09e75b74c20c1abdc7b9a9a47d9e673/readthedocs/projects/tasks.py#L254-L265

Closes #4590

### CODE REVIEW

File: readthedocs/vcs_support/base.py [hunk 1]
```diff
@@ -102,7 +103,13 @@ def run(self, *cmd, **kwargs):
             'shell': False,
         })
 
-        build_cmd = self.environment.run(*cmd, **kwargs)
+        try:
+            build_cmd = self.environment.run(*cmd, **kwargs)
+        except BuildEnvironmentWarning as e:
```
Comment (humitos) [inline readthedocs/vcs_support/base.py]: It is supposed that we are not logging `BuildEnvironmentWarning` already (https://github.com/readthedocs/readthedocs.org/blob/70ca7c737a3db2539160e74f8ababaaf4743f30a/readthedocs/projects/tasks.py#L286). I'm not sure to follow why catching this exception and re-raising it as `RepositoyError` will avoid the log in Sentry.

I feel confused here :confused:

Comment (stsewd) [reply readthedocs/vcs_support/base.py]: Yes, but that isn't the cause. We are logging it manually here 

https://github.com/readthedocs/readthedocs.org/blob/00ab116ec09e75b74c20c1abdc7b9a9a47d9e673/readthedocs/projects/tasks.py#L254-L265

That try...except block only captures RepositoryError and LockTimeout, everything else is logged

Comment (humitos) [reply readthedocs/vcs_support/base.py]: Gotcha! Do you know why the `sync_repo` is not ran inside the context manager that manages all the exceptions in the `__exit__` as we do run all the rest of the commands?

Comment (stsewd) [reply readthedocs/vcs_support/base.py]: It runs inside one https://github.com/readthedocs/readthedocs.org/blob/83e31f87c46845d11bfe6c7de8a417a6fdfb0c3f/readthedocs/projects/tasks.py#L449-L456

But again, that's not the problem. We still log the exception to sentry manually.

Comment (humitos) [reply readthedocs/vcs_support/base.py]: I'm talking about this one which does not seem to be ran inside a context manager:

https://github.com/readthedocs/readthedocs.org/blob/83e31f87c46845d11bfe6c7de8a417a6fdfb0c3f/readthedocs/projects/tasks.py#L243

Comment (stsewd) [reply readthedocs/vcs_support/base.py]: Not really sure, we don't record that action as a build, so we don't need the all the stuff from the context manager. It doesn't seen related to this anyway. Both calls use the default env https://github.com/readthedocs/readthedocs.org/blob/83e31f87c46845d11bfe6c7de8a417a6fdfb0c3f/readthedocs/vcs_support/base.py#L67-L67 and don't use the context manager.

Comment (humitos) [reply readthedocs/vcs_support/base.py]: Let's merge and see what happen after the next deploy :)

**Comment by stsewd** [comment]
@humitos errors from that place disappeared, but now it's logging from another place (where we log manually too)

https://github.com/readthedocs/readthedocs.org/blob/557c875fa4d74e1cb69146e005375595f80adf91/readthedocs/doc_builder/environments.py#L580-L623

Looks it can be fixed by adding the exception in this list

https://github.com/readthedocs/readthedocs.org/blob/557c875fa4d74e1cb69146e005375595f80adf91/readthedocs/doc_builder/environments.py#L536-L542

**Comment by humitos** [comment]
Good!

It is supposed the WARNING_EXCEPTIONS is not needed anymore after we added the `throws` argument in the celery task. Actually, there should be an issue saying that we have to remove it.

The right place to add the exception class is https://github.com/readthedocs/readthedocs.org/blob/30e495504911d0720558a6b307c9cd05a7dbfffb/readthedocs/projects/tasks.py#L273-L292

**Comment by stsewd** [comment]
@humitos the class is already on the throws arg. We should remove the code that manually logs to sentry.

**Pull request merged**